res/controllers/Hercules-DJControl-Inpulse-300-script.js

```diff
@@ -529,27 +523,23 @@ DJCi300.updateToneplayLED = function(value, group, _control) {
 
 // Functions that connect and disconnect slicer functions to Mixxx callbacks, respectively
 DJCi300.connectSlicerFunctions = function(deck) {
+    const index = deck - 1;
     if (deck === 1) {
-        slicerSizeConnection1 = engine.makeConnection("[Channel1]", "beatloop_size", DJCi300.slicerChangeSize);
-        slicerTrackConnection1 = engine.makeConnection("[Channel1]", "LoadSelectedTrack", DJCi300.slicerLoadTrack);
-        slicerBeatConnection1 = engine.makeConnection("[Channel1]", "beat_distance", DJCi300.slicerCountBeat);
+        DJCi300.slicerConnections[size][index] = engine.makeConnection("[Channel1]", "beatloop_size", DJCi300.slicerChangeSize);
+        DJCi300.slicerConnections[load][index] = engine.makeConnection("[Channel1]", "LoadSelectedTrack", DJCi300.slicerLoadTrack);
+        DJCi300.slicerConnections[beat][index] = engine.makeConnection("[Channel1]", "beat_distance", DJCi300.slicerCountBeat);
```

Comment:
this will look up strings in a variable called `size` (which is undefined). Instead do 
```suggestion
        DJCi300.slicerConnections.size[index] = engine.makeConnection("[Channel1]", "beatloop_size", DJCi300.slicerChangeSize);
        DJCi300.slicerConnections.load[index] = engine.makeConnection("[Channel1]", "LoadSelectedTrack", DJCi300.slicerLoadTrack);
        DJCi300.slicerConnections.beat[index] = engine.makeConnection("[Channel1]", "beat_distance", DJCi300.slicerCountBeat);
```